src/main/java/oss/fosslight/service/impl/SystemConfigurationServiceImpl.java

```diff
@@ -379,6 +380,14 @@ public Map<String, Object> ldapTestConnection(Map<String, Object> configurationM
 
 
 		if(loginDetailMap != null) {
+			resAddress.append(loginDetailMap.get(CoConstDef.CD_LDAP_PROTOCOL));
+			resAddress.append("://");
+			resAddress.append(loginDetailMap.get(CoConstDef.CD_LDAP_URL));
+			resAddress.append(":");
+			resAddress.append(loginDetailMap.get(CoConstDef.CD_LDAP_PORT));
+			resAddress.append(" ");
+			resAddress.append(loginDetailMap.get(CoConstDef.CD_LDAP_SEARCH_ID));
+

```

Comment:
I think using UriComponentsBuilder is more readable than StringBuilder. Here is a guide using UriComponentsBuilder. https://www.baeldung.com/spring-uricomponentsbuilder